### ISSUE TITLE
How can I add a env condition in a component with locals variable ?

### ISSUE THREAD

**Issue by jbtheard** [maizzle/framework#396]
Hi there,

How can I access page.env in a condition when that condition is within a component with local variable ?

myComponent.html
```
<div >
  <if condition="page.env === 'local'"> 
    <p>{{content.local}}</p>
   </if>
 <elseif condition="page.env === 'production'">
    <p>{{content.prod}}</p>
  </elseif> 
</div>
```

used in my template.html 
```
<component src="src/components/myComponent.html" locals='{"content":{"local": "Local", "prod": "Production"}'></component>
```

I understand that when passing _locals_ variable, page.env isn't accessible anymore and template fails to compile.
I have also tried to pass env as an extra key to locals but couldn't figure out how to access it in a condition tag

Any idea?

- Maizzle Version: client 1.3.0 / Framework 3.1.2
- Node.js Version: 15.4.0

**Comment by cossssmin** [comment]
Unfortunately due to how `posthtml-modules` works, this is not possible at the moment - the only way to access the `page` object in a Component is to not include the `locals=''` attribute on it.

A workaround would be to define your data that you want to pass into the Component right in your config - it can be done either manually, or programatically through [events](https://maizzle.com/docs/events/).

**Comment by cossssmin** [comment]
Opened in https://github.com/posthtml/posthtml-modules/issues/56, I think we could do something about it.

**Comment by jbtheard** [comment]
Thanks for your reply @cossssmin,

I'm not quite sure the workaround would work as it's a reusable component (say a button) that can take different values (hence using _locals_).

A bit of context: I'd like to develop a multi language template (with reusable components) to be used with SendGrid (I don't want to use different templates). As per [their doc](https://sendgrid.com/docs/for-developers/sending-email/using-handlebars/#multiple-languages) we handle languages with a handlebar condition. 
Maizzle works fine handling these tags but visually the rendered page is a bit ugly (with all languages + conditions displayed). 

In dev mode I'd like to display a single language but compile with all languages when building template.

If you have any tip suggestion on how to implement that; I'm all ears :)

**Comment by cossssmin** [comment]
See https://github.com/posthtml/posthtml-modules/pull/57

I've already tested with Maizzle locally, works as expected. I'll bump `posthtml-modules` here once it's updated to include that feature.

**Comment by cossssmin** [comment]
> In dev mode I'd like to display a single language but compile with all languages when building template.
> 
> If you have any tip suggestion on how to implement that; I'm all ears :)

If I get what you want to do right, you can just define `content` in `config.js`, then override it in `config.production.js`. These two get merged to create a template-specific config - see [build process docs](https://maizzle.com/docs/build-process/). 

No need for `content.local` or `content.production`, the template will just use what's inside `content` as it's defined in the config for the environment that you ran the command for.

Hope that makes sense :)

**Comment by cossssmin** [comment]
@jbtheard update to [`v3.1.3`](https://github.com/maizzle/framework/releases/tag/v3.1.3) and try again, it should be working now.

**Comment by boreenberg** [comment]
> @jbtheard update to [`v3.1.3`](https://github.com/maizzle/framework/releases/tag/v3.1.3) and try again, it should be working now.

@cosssmin When updating to 3.1.3 posthtml-modules seems to be  bumped to 0.7.2 - is this by-design. My nested components render really poorly with 3.1.3 - also on higher level in hierarchy. With 3.10. the issues are only on deeper levels, more random and linked to use of locals. If I set modules to 0.6.2, rendering under 3.1.3 seems similar to 3.1.0. Hope it makes sense - thanks!

**Comment by cossssmin** [comment]
Yes, we've bumped `posthtml-modules` to the latest version and it now evaluates expressions inside components, so that you can use both `page`-level variables and locals passed through the attribute.

Is it possible for you to share a repo that reproduces that issue? Either public or private is fine.

**Comment by boreenberg** [comment]
> Yes, we've bumped `posthtml-modules` to the latest version and it now evaluates expressions inside components, so that you can use both `page`-level variables and locals passed through the attribute.
> 
> Is it possible for you to share a repo that reproduces that issue? Either public or private is fine.

@cossssmin thanks - i've sent you an invite to repo

**Comment by jbtheard** [comment]
Sorry I couldn't test earlier but I confirm it 100% solves my case.
That's great as to me it opens ways for component based development in emails 🚀 🚀 🚀 

Thanks for this lib and your awesome support!

**Comment by cossssmin** [comment]
@jbtheard awesome, happy to hear that! 🥳

@boreenberg thanks for the invite, I see you're heavily using component nesting with passing locals or resetting them to `{}` when needed. Interesting setup and definitely not a common use case, but this is a great example that can help us improve.

I thought this was about speed, but in fact it relates to #369 (see [this comment](https://github.com/maizzle/framework/issues/369#issuecomment-726224861)), which hasn't yet been resolved in `posthtml-modules` (https://github.com/posthtml/posthtml-modules/issues/51). Until we have a fix for that, I suggest sticking with 3.1.0.

Since this issue was about something else, let's keep the conversation going in #369.
